training/src/main/scala/com/airbnb/aerosolve/training/ForestTrainer.scala

```diff
@@ -43,6 +43,15 @@ object ForestTrainer {
         .map(x => Util.flattenFeature(x.example(0)))
         .filter(x => x.contains(rankKey))
         .coalesce(numTrees, true)
+
+    val ex = examples.take(1)(0)
+    val numFeatures = LinearRankerUtils.getNumFeatures(ex, rankKey)
+    val maxFeatures : Int = Try(config.getString(key + ".max_features")).getOrElse("all") match {
+      case "all" => Int.MaxValue
+      case "sqrt" => math.sqrt(numFeatures).ceil.toInt
+      case "log2" => math.max(1, (math.log(numFeatures) / math.log(2)).ceil.toInt)
```

Comment:
because the `ForestTrainer` currently directly calls the buildTree function in decisionTree training. I think it is better to have `buildTree` function just take a int argument so it works independent of what the config value could be for different models. The `ForestTrainer` and DecisionTreeTrainer currently have the same logic in term of parsing the max_features config but they can be different later on. Let me know if makes sense to you. @jq